feat(docs): restructure content and fix sidebar layout

vivekjain23 opened the pull request:
# Description of change

This PR introduces changes to the sidebar layout and reorganizes documentation content.

## Links to any relevant issues

fixes #7348

Closes #5522 

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### Release Notes

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

vivekjain23 committed:
Feat(Docs): Restructure docs:- created users moved wallet

vivekjain23 committed:
fix(docs): removed redundant wallet docs from about iota

vivekjain23 committed:
feat(docs): restructure docs moved ts-sdk into developer section

vivekjain23 committed:
feat(docs):  moved ts-sdk into developer section and referencing changes accordingly

vivekjain23 committed:
feat(docs):  moved iota-evm into developer section and referencing respectively

vivekjain23 committed:
feat(docs): changes made in the scripts, added navbar icons and referencing links updated

vivekjain23 committed:
feat(docs): docs restructure changes and updated the structure changes referencing respectively

vivekjain23 committed:
feat(docs): restructure docs moved iota identity into developer section and update references

vivekjain23 committed:
feat(docs): update referencing links and correct IOTA Identity references

vivekjain23 committed:
updated correct referencing links

vivekjain23 committed:
updated the collapsed inside sidebar

Dr-Electron commented:
Looks good to me. Can we move References under Developers? I think having it in the top bar AND in the Developers sidebar is confusing (I know it is different content. But still)

vivekjain23 committed:
fix(docs): moved references in developer section

vivekjain23 committed:
Merge branch 'develop' into devx/restructure-docs-sidebar-layout

Thoralf-M commented:
What about https://github.com/iotaledger/iota/issues/4702 ? This seems still inconsistent

Dr-Electron committed:
Fix generated docs slug

Dr-Electron commented on docs/content/developer/stardust/stardust-migration.mdx:
```diff
@@ -20,12 +20,12 @@ The Stardust ledger state, and hence the assets and tokens in the network, are r
 
 With the introduction of the object-based Move ledger in IOTA Rebased, transactions no longer consume and create outputs
-but [`Move Objects`](../iota-101/objects/object-model.mdx) instead.
+but [`Move Objects`](../iota-101/objects/object-model.mdx) instead.x
```
```suggestion
but [`Move Objects`](../iota-101/objects/object-model.mdx) instead.
```

Dr-Electron commented on docs/content/sidebars/users.js:
```diff
@@ -0,0 +1,78 @@
+    },
+];
+
+module.exports = users;
```
```suggestion
module.exports = users;

```

Dr-Electron commented on docs/content/sidebars/ts-sdk.js:
```diff
@@ -131,16 +132,16 @@ const tsSDK = [
 ];
 
-module.exports = tsSDK;
+module.exports = tsSDK;
```
```suggestion
module.exports = tsSDK;

```

Dr-Electron committed:
Temporary fix broken links

Dr-Electron committed:
Fix sidebars

Dr-Electron commented on docs/site/scripts/get-iota-evm-references.sh:
```diff
@@ -1,16 +1,19 @@
 # Create temporary directory to work in
-mkdir tmp
-cd tmp
+mkdir -p tmp
+cd tmp || exit
```
```suggestion
cd tmp
```

Dr-Electron replied:
```suggestion
mkdir tmp
cd tmp
```

Dr-Electron commented on docs/site/scripts/get-iota-evm-references.sh:
```diff
@@ -1,16 +1,19 @@
 
 # Return to root and cleanup
-cd -
+cd - || exit
```
```suggestion
cd -
```

Dr-Electron commented on docs/site/src/css/custom.css:
```diff
@@ -643,3 +643,48 @@ h4 {
+
+.navbar__item.navbar-icon-users::before {
+  content: "\f0c0"; 
+}
```
```suggestion
}

```

Dr-Electron committed:
fix remaining paths and cleanup references

Dr-Electron committed:
Apply suggestions from code review

Dr-Electron committed:
Fix one more thing

Dr-Electron committed:
Add redirects

Dr-Electron committed:
Merge branch 'develop' into devx/restructure-docs-sidebar-layout

Dr-Electron committed:
revert: remove non-docs changes from docs restructure PR

This PR should only contain documentation restructuring changes.
Non-docs changes have been moved to devx/non-docs-changes branch.

Dr-Electron committed:
Move idendity refs

Dr-Electron committed:
Fix small typo

Dr-Electron committed:
Fix broken links

Dr-Electron committed:
Fix sripts

Dr-Electron committed:
Fix gitignore

Dr-Electron commented on docs/content/developer/iota-identity/references/.gitignore:
```diff
@@ -0,0 +1 @@
+wasm/**
```
```suggestion
wasm/**

```

Dr-Electron committed:
Update docs/content/developer/iota-identity/references/.gitignore
